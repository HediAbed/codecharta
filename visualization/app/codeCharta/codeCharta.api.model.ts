import {
	AttributeTypes,
	AttributeTypeValue,
	CodeMapNode,
	Edge,
	MarkedPackage,
	MetricStatisticsItem,
	RecursivePartial,
	Settings
} from "./codeCharta.model"

export interface ExportCCFile {
	projectName: string
	apiVersion: string
	fileChecksum: string
	nodes: CodeMapNode[]
	attributeTypes?: AttributeTypes | OldAttributeTypes
	edges?: Edge[]
	markedPackages?: MarkedPackage[]
	blacklist?: ExportBlacklistItem[]
	metricStatistics?: ExportMetricStatisticsItem[]
}

export interface ExportBlacklistItem {
	path: string
	type: ExportBlacklistType
}

export type ExportMetricStatisticsItem = MetricStatisticsItem

export enum ExportBlacklistType {
	hide = "hide",
	exclude = "exclude"
}

export enum APIVersions {
	ZERO_POINT_ONE = "0.1",
	ONE_POINT_ZERO = "1.0",
	ONE_POINT_ONE = "1.1",
	ONE_POINT_TWO = "1.2"
}

export interface ExportScenario {
	name: string
	settings: RecursivePartial<Settings>
}

export interface OldAttributeTypes {
	nodes?: [{ [key: string]: AttributeTypeValue }?]
	edges?: [{ [key: string]: AttributeTypeValue }?]
}
