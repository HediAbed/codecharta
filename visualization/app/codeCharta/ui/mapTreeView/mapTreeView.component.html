<map-tree-view-level-component
	ng-if="$ctrl._viewModel.rootNode"
	node="$ctrl._viewModel.rootNode"
	depth="0"
></map-tree-view-level-component>
