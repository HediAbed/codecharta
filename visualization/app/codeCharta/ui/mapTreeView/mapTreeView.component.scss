map-tree-view-component {
	.tree-element-0 {
		margin-left: 3px;
		margin-top: 10px;
	}

	.tree-root {
		overflow-y: auto;
		max-height: 400px;

		div {
			margin: 0;
			padding: 0;
		}

		.tree-element-children {
			margin-left: 22px;
		}
	}

	.tree-leaf .tree-element-label {
		cursor: default;
	}

	.tree-element-label {
		cursor: pointer;

		&:focus {
			outline: none;
		}

		&:hover,
		&.hovered {
			background-color: #efefef;
			border-radius: 2px;
		}

		.unary-number {
			color: #a3a3a3;
			font-size: 0.8em;
			margin-top: 1px;
		}

		.node-type-icon {
			display: inline-block;
			width: 14px;
			margin-left: 2px;
		}
	}
}
