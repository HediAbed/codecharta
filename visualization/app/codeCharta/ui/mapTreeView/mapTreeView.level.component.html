<div ng-if="!$ctrl.isBlacklisted($ctrl._viewModel.node)" class="{{::$ctrl._viewModel.depth?'':'tree-root'}} tree-element-{{::$ctrl._viewModel.depth?$ctrl._viewModel.depth:0}} {{::$ctrl.isLeaf()?'tree-leaf':'tree-parent'}}">

    <div class="tree-element-label-{{::$ctrl._viewModel.depth?$ctrl._viewModel.depth:0}} tree-element-label {{$ctrl._viewModel.isHoveredInCodeMap ? 'hovered':''}}" ng-mouseenter="$ctrl.onMouseEnter()" ng-mouseleave="$ctrl.onMouseLeave()" ng-right-click="$ctrl.onRightClick($event)">

        <span ng-click="$ctrl.onFolderClick()">
            <span role="img" ng-if="!$ctrl.isLeaf() && $ctrl._viewModel.collapsed" class="fa fa-folder cursor-pointer" ng-style="{'color': $ctrl.getMarkingColor() }" aria-hidden="true"></span>
            <span role="img" ng-if="!$ctrl.isLeaf() && !$ctrl._viewModel.collapsed" class="fa fa-folder-open cursor-pointer" ng-style="{'color': $ctrl.getMarkingColor() }" aria-hidden="true"></span>
            <span role="img" ng-if="$ctrl.isLeaf()" class="fa fa-file-o" aria-hidden="true"></span>
        </span>

        <span class="cursor-pointer" ng-click="$ctrl.onEyeClick()">
            <span role="img" ng-if="$ctrl._viewModel.node.visible" class="fa fa-eye" aria-hidden="true"></span>
            <span role="img" ng-if="!$ctrl._viewModel.node.visible" class="fa fa-eye-slash" aria-hidden="true"></span>
        </span>

        <span class="cursor-pointer {{$ctrl.isSearched($ctrl._viewModel.node)?'angular-green':''}}" ng-click="$ctrl.onLabelClick()">
            {{::$ctrl._viewModel.node.name}}
        </span>
    </div>

    <div class="tree-element-children tree-element-children-{{::$ctrl._viewModel.depth?$ctrl._viewModel.depth:0}}">

        <div ng-if="!$ctrl._viewModel.collapsed"
             ng-repeat="node in $ctrl._viewModel.node.children | orderBy: [$ctrl.sortByFolder,'name'] ">
            <map-tree-view-level-component node="::node" depth="::$ctrl.depth+1"></map-tree-view-level-component>
        </div>

    </div>

</div>
