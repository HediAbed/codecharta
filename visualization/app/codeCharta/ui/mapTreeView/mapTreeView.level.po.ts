import { Page } from "puppeteer"

export class MapTreeViewLevelPageObject {
	constructor(private page: Page) {}

	public async openContextMenu() {
		await this.page.click("[id='/root/ParentLeaf']")
		await this.page.click("[id='/root/ParentLeaf/smallLeaf.html']", { button: "right" })
	}

	public async hoverNode() {
		await this.page.click("[id='/root/ParentLeaf']")
		await this.page.hover("[id='/root/ParentLeaf/smallLeaf.html']")
	}

	public async nodeExists() {
		return !!(await this.page.$("[id='/root/ParentLeaf/smallLeaf.html']"))
	}
}
