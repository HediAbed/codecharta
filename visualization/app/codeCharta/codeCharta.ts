"use strict";

import "./core/core.module";
import "./ui/ui";

import {codeChartaComponent} from "./codeCharta.component";

import angular from "angular";
import { CodeChartaService } from "./codeCharta.service";
import _ from "lodash";

angular.module(
    "app.codeCharta",
    ["app.codeCharta.core", "app.codeCharta.ui"]
);

angular.module("app.codeCharta").component(
    codeChartaComponent.selector,
    codeChartaComponent
).service(
    _.camelCase(CodeChartaService.name),
    CodeChartaService
);


