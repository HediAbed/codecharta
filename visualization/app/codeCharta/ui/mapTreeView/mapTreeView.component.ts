import { IRootScopeService, ITimeoutService } from "angular"
import { CodeMapNode, NodeType, SortingOption } from "../../codeCharta.model"
import { CodeMapPreRenderService, CodeMapPreRenderServiceSubscriber } from "../codeMap/codeMap.preRender.service"
import { StoreService } from "../../state/store.service"
import {
	SortingDialogOptionService,
	SortingDialogOptionSubscriber
} from "../../state/store/dynamicSettings/sortingDialogOption/sortingDialogOption.service"
import {
	SortingOrderAscendingService,
	SortingOrderAscendingSubscriber
} from "../../state/store/appSettings/sortingOrderAscending/sortingOrderAscending.service"
import { setSortingDialogOption } from "../../state/store/dynamicSettings/sortingDialogOption/sortingDialogOption.actions"
import { setSortingOrderAscending } from "../../state/store/appSettings/sortingOrderAscending/sortingOrderAscending.actions"

export class MapTreeViewController
	implements CodeMapPreRenderServiceSubscriber, SortingDialogOptionSubscriber, SortingOrderAscendingSubscriber {
	private _viewModel: {
		rootNode: CodeMapNode
	} = {
		rootNode: null
	}

	/* @ngInject */
	constructor(private $rootScope: IRootScopeService, private $timeout: ITimeoutService, private storeService: StoreService) {
		CodeMapPreRenderService.subscribe(this.$rootScope, this)
		SortingDialogOptionService.subscribe(this.$rootScope, this)
		SortingOrderAscendingService.subscribe(this.$rootScope, this)
	}

	public onSortingDialogOptionChanged(sortingDialogOption: SortingOption) {
		if (sortingDialogOption === SortingOption.NUMBER_OF_FILES) {
			this._viewModel.rootNode = this.applySortOrderChange(
				this._viewModel.rootNode,
				(a, b) => b.attributes["unary"] - a.attributes["unary"],
				false
			)
			return
		}
		this._viewModel.rootNode = this.applySortOrderChange(this._viewModel.rootNode, (a, b) => (b.name > a.name ? 0 : 1), false)
	}

	public onSortingOrderAscendingChanged(sortingOrderAscending: boolean) {
		this._viewModel.rootNode = this.applySortOrderChange(this._viewModel.rootNode, null, true)
	}

	public applySortOrderChange(node: CodeMapNode, compareFn: (a: CodeMapNode, b: CodeMapNode) => number, reverse: boolean) {
		if (!node) {
			return
		}
		for (let i = 0; i < node.children.length; i++) {
			if (node.children[i].type === NodeType.FOLDER) {
				node.children[i] = this.applySortOrderChange(node.children[i], compareFn, reverse)
			}
		}
		if (!reverse) {
			node.children = this.groupFilesAndFolders(node, compareFn)
		} else {
			node.children.reverse()
		}
		return node
	}

	private groupFilesAndFolders(node: CodeMapNode, compareFn: (a: CodeMapNode, b: CodeMapNode) => number) {
		const folders = node.children.filter(node => node.type === NodeType.FOLDER)
		const files = node.children.filter(node => node.type === NodeType.FILE)
		folders.sort(compareFn)
		files.sort(compareFn)
		return folders.concat(files)
	}

	public onRenderMapChanged(map: CodeMapNode) {
		if (map === this._viewModel.rootNode) {
			// needed to prevent flashing since event is triggered 4 times
			return
		}
		this._viewModel.rootNode = map
		this.synchronizeAngularTwoWayBinding()

		this.storeService.dispatch(setSortingDialogOption(this.storeService.getState().dynamicSettings.sortingDialogOption))
		if (this.storeService.getState().appSettings.sortingOrderAscending) {
			this.storeService.dispatch(setSortingOrderAscending(this.storeService.getState().appSettings.sortingOrderAscending))
		}
	}

	private synchronizeAngularTwoWayBinding() {
		this.$timeout(() => {})
	}
}

export const mapTreeViewComponent = {
	selector: "mapTreeViewComponent",
	template: require("./mapTreeView.component.html"),
	controller: MapTreeViewController
}
