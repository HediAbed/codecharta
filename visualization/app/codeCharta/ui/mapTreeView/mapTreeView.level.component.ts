import {SettingsService} from "../../core/settings/settings.service";
import {IRootScopeService, IAngularEvent} from "angular";
import {NodeContextMenuController} from "../nodeContextMenu/nodeContextMenu.component";
import {CodeMapActionsService} from "../codeMap/codeMap.actions.service";
import {CodeMapUtilService} from "../codeMap/codeMap.util.service";
import {AngularColors} from "../codeMap/rendering/renderSettings";
import { CodeMapMouseEventServiceSubscriber, CodeMapBuildingTransition, CodeMapMouseEventService } from "../codeMap/codeMap.mouseEvent.service";
import { CodeMapBuilding } from "../codeMap/rendering/codeMapBuilding";
import { CodeMapNode, BlacklistType } from "../../codeCharta.model";

export interface MapTreeViewHoverEventSubscriber {
    onShouldHoverNode(node: CodeMapNode);
    onShouldUnhoverNode(node: CodeMapNode);
}

export class MapTreeViewLevelController implements CodeMapMouseEventServiceSubscriber{

    public _isHoveredInCodeMap: boolean = false;
    public node: CodeMapNode = null;
    public depth: number = 0;
    public collapsed: boolean = true;
    public angularGreen: string = AngularColors.green;

    /* @ngInject */
    constructor(
        private $rootScope: IRootScopeService,
        private codeMapActionsService: CodeMapActionsService,
        private settingsService: SettingsService
    ) {
        CodeMapMouseEventService.subscribe(this.$rootScope, this);
    }

    public getFolderColor() {
        if(!this.node) {
            return "#000";
        }
        return this.node.markingColor ? "#" + this.node.markingColor.substr(2) : "#000";
    }

    public onBuildingHovered(data: CodeMapBuildingTransition, event: IAngularEvent) {
        if(data.to && data.to.node && this.node && this.node.path && data.to.node.path === this.node.path) {
            this._isHoveredInCodeMap = true;
        } else {
            this._isHoveredInCodeMap = false;
        }
    }
    public onBuildingSelected(data: CodeMapBuildingTransition, event: IAngularEvent) {
        // unused
    }
    public onBuildingRightClicked(building: CodeMapBuilding, x: number, y: number, event: IAngularEvent) {
        // unused
    }

    public onMouseEnter() {
        this.$rootScope.$broadcast("should-hover-node", this.node);
    }

    public onMouseLeave() {
        this.$rootScope.$broadcast("should-unhover-node", this.node);
    }

    public onRightClick($event) {
        NodeContextMenuController.broadcastHideEvent(this.$rootScope);
        NodeContextMenuController.broadcastShowEvent(this.$rootScope, this.node.path, this.node.type, $event.clientX, $event.clientY);
    }

    public onFolderClick() {
        this.collapsed = !this.collapsed;
    }

    public onLabelClick() {
        this.codeMapActionsService.focusNode(this.node);
    }

    public onEyeClick() {
        this.codeMapActionsService.toggleNodeVisibility(this.node);
    }

    public isLeaf(node: CodeMapNode = this.node): boolean {
        return !(node && node.children && node.children.length > 0);
    }

    public isBlacklisted(node: CodeMapNode): boolean {
        if (node != null) {
            return CodeMapUtilService.isBlacklisted(node, this.settingsService.settings.mapSettings.blacklist, BlacklistType.exclude)
        }
        return false;
    }

    public isSearched(node: CodeMapNode): boolean {
        if (node != null && this.settingsService.settings.mapSettings.searchedNodePaths) {
            return this.settingsService.settings.mapSettings.searchedNodePaths.filter(path =>
                path == node.path).length > 0;
        }
        return false;
    }

    public sortByFolder(node: CodeMapNode) {
        return (node && node.children && node.children.length > 0) ? 1 : 0;
    }

    public static subscribeToHoverEvents($rootScope: IRootScopeService, subscriber: MapTreeViewHoverEventSubscriber){
        $rootScope.$on("should-hover-node", (event, args)=>subscriber.onShouldHoverNode(args));
        $rootScope.$on("should-unhover-node", (event, args)=>subscriber.onShouldUnhoverNode(args));
    }

}

export const mapTreeViewLevelComponent = {
    selector: "mapTreeViewLevelComponent",
    template: require("./mapTreeView.level.component.html"),
    controller: MapTreeViewLevelController,
    bindings: {
        node: "<",
        depth: "<"
    }
};




