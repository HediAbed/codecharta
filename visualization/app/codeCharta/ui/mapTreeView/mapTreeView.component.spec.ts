import "./mapTreeView.module"
import { MapTreeViewController } from "./mapTreeView.component"
import { CodeMapPreRenderService } from "../codeMap/codeMap.preRender.service"
import { IRootScopeService, ITimeoutService } from "angular"
import { instantiateModule, getService } from "../../../../mocks/ng.mockhelper"
import { CodeMapNode } from "../../codeCharta.model"
import {
	VALID_NODE_WITH_MULTIPLE_FOLDERS,
	VALID_NODE_WITH_MULTIPLE_FOLDERS_SORTED_BY_NAME,
	VALID_NODE_WITH_MULTIPLE_FOLDERS_SORTED_BY_UNARY,
	VALID_NODE_WITH_PATH
} from "../../util/dataMocks"
import _ from "lodash"
import { StoreService } from "../../state/store.service"
import { SortingDialogOptionService } from "../../state/store/dynamicSettings/sortingDialogOption/sortingDialogOption.service"
import { SortingOrderAscendingService } from "../../state/store/appSettings/sortingOrderAscending/sortingOrderAscending.service"

describe("MapTreeViewController", () => {
	let mapTreeViewController: MapTreeViewController
	let $rootScope: IRootScopeService
	let $timeout: ITimeoutService
	let storeService = getService<StoreService>("storeService")
	let map: CodeMapNode

	beforeEach(() => {
		restartSystem()
		rebuildController()
	})

	function restartSystem() {
		instantiateModule("app.codeCharta.ui.mapTreeView")

		$rootScope = getService<IRootScopeService>("$rootScope")
		$timeout = getService<ITimeoutService>("$timeout")
		storeService = getService<StoreService>("storeService")

		map = _.cloneDeep(VALID_NODE_WITH_PATH)
	}

	function rebuildController() {
		mapTreeViewController = new MapTreeViewController($rootScope, $timeout, storeService)
	}

	describe("constructor", () => {
		it("should subscribe to CodeMapPreRenderService", () => {
			CodeMapPreRenderService.subscribe = jest.fn()

			rebuildController()

			expect(CodeMapPreRenderService.subscribe).toHaveBeenCalledWith($rootScope, mapTreeViewController)
		})

		it("should subscribe to SortingDialogOptionService", () => {
			SortingDialogOptionService.subscribe = jest.fn()

			rebuildController()

			expect(SortingDialogOptionService.subscribe).toHaveBeenCalledWith($rootScope, mapTreeViewController)
		})
		it("should subscribe to SortingOrderAscendingService", () => {
			SortingOrderAscendingService.subscribe = jest.fn()

			rebuildController()

			expect(SortingOrderAscendingService.subscribe).toHaveBeenCalledWith($rootScope, mapTreeViewController)
		})
	})

	describe("applySort", () => {
		it("should sort by unary", () => {
			let validNodeSorted = VALID_NODE_WITH_MULTIPLE_FOLDERS
			let sortedNode = VALID_NODE_WITH_MULTIPLE_FOLDERS_SORTED_BY_UNARY
			let compareFn = (a, b) => b.attributes["unary"] - a.attributes["unary"]
			let result = mapTreeViewController.applySortOrderChange(validNodeSorted, compareFn, false)
			expect(result).toEqual(sortedNode)
		})

		it("should sort by name", () => {
			let validNodeSorted = VALID_NODE_WITH_MULTIPLE_FOLDERS
			let sortedNode = VALID_NODE_WITH_MULTIPLE_FOLDERS_SORTED_BY_NAME
			let compareFn = (a, b) => (b.name > a.name ? 1 : 0)
			let result = mapTreeViewController.applySortOrderChange(validNodeSorted, compareFn, false)
			console.log(result)
			expect(result).toEqual(sortedNode)
		})
	})

	describe("onSortingOrderAscendingChanged", () => {
		it("", () => {})
	})

	describe("onSortingDialogOptionChanged", () => {
		it("", () => {})
	})

	describe("applySort", () => {
		it("", () => {})
	})

	/*	it("should sort folder", () => {
			mapTreeViewLevelController["node"] = CodeMapHelper.getCodeMapNodeFromPath(
				"/root/Parent Leaf",
				NodeType.FOLDER,
				VALID_NODE_WITH_PATH
			)

			const result = mapTreeViewLevelController.sortByFolder(mapTreeViewLevelController["node"])

			expect(result).toBe(1)
		})

*/

	describe("onRenderMapChanged", () => {
		it("should update viewModel.rootNode after timeout", () => {
			mapTreeViewController["_viewModel"] = { rootNode: null }
			mapTreeViewController["onRenderMapChanged"](map)

			//mapTreeViewController.onRenderMapChanged(map)
			$timeout.flush(100)

			expect(mapTreeViewController["_viewModel"].rootNode).toBe(map)
		})
	})
	describe("onSortingDialogOptionChanged", () => {
		it("should sort the nodes according to name, splitting into Folders and Files while preserving tree structure", () => {})
	})
})
