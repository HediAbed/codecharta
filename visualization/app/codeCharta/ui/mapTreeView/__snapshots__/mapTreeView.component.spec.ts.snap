// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapTreeViewController onRenderMapChanged should clone new map before updating viewModel.rootNode to prevent unwanted changes in original map 1`] = `
Object {
  "attributes": Object {},
  "children": Array [
    Object {
      "attributes": Object {},
      "children": Array [],
      "isExcluded": false,
      "isFlattened": false,
      "name": "Folder1",
      "type": "Folder",
    },
    Object {
      "attributes": Object {},
      "children": Array [],
      "isExcluded": false,
      "isFlattened": false,
      "name": "Folder2",
      "type": "Folder",
    },
    Object {
      "attributes": Object {},
      "children": Array [
        Object {
          "attributes": Object {
            "functions": 100,
            "mcc": 100,
            "rloc": 30,
          },
          "isExcluded": false,
          "isFlattened": false,
          "name": "small leaf",
          "type": "File",
        },
      ],
      "isExcluded": false,
      "isFlattened": false,
      "name": "Folder3",
      "type": "Folder",
    },
    Object {
      "attributes": Object {
        "functions": 10,
        "mcc": 1,
        "rloc": 100,
      },
      "isExcluded": false,
      "isFlattened": false,
      "link": "http://www.google.de",
      "name": "big leaf",
      "type": "File",
    },
  ],
  "isExcluded": false,
  "isFlattened": false,
  "name": "root",
  "type": "Folder",
}
`;

exports[`MapTreeViewController onRenderMapChanged should update viewModel.rootNode after timeout and sort 1`] = `
Object {
  "attributes": Object {
    "mcc": 111,
    "rloc": 200,
  },
  "children": Array [
    Object {
      "attributes": Object {
        "mcc": 110,
        "rloc": 100,
      },
      "children": Array [
        Object {
          "attributes": Object {
            "functions": 1000,
            "mcc": 10,
            "rloc": 70,
          },
          "edgeAttributes": Object {
            "average_commits": Object {
              "incoming": 0,
              "outgoing": 0,
            },
            "pairing_rate": Object {
              "incoming": 0,
              "outgoing": 0,
            },
          },
          "id": 4,
          "isExcluded": false,
          "isFlattened": false,
          "name": "other small leaf",
          "path": "/root/Parent Leaf/other small leaf",
          "type": "File",
        },
        Object {
          "attributes": Object {
            "functions": 100,
            "mcc": 100,
            "rloc": 30,
          },
          "edgeAttributes": Object {
            "average_commits": Object {
              "incoming": 0,
              "outgoing": 0,
            },
            "pairing_rate": Object {
              "incoming": 0,
              "outgoing": 0,
            },
          },
          "id": 3,
          "isExcluded": false,
          "isFlattened": false,
          "name": "small leaf",
          "path": "/root/Parent Leaf/small leaf",
          "type": "File",
        },
      ],
      "descendants": 2,
      "edgeAttributes": Object {
        "average_commits": Object {
          "incoming": 0,
          "outgoing": 0,
        },
        "pairing_rate": Object {
          "incoming": 0,
          "outgoing": 0,
        },
      },
      "id": 2,
      "isExcluded": false,
      "isFlattened": false,
      "name": "Parent Leaf",
      "path": "/root/Parent Leaf",
      "type": "Folder",
    },
    Object {
      "attributes": Object {
        "functions": 10,
        "mcc": 1,
        "rloc": 100,
      },
      "edgeAttributes": Object {
        "average_commits": Object {
          "incoming": 0,
          "outgoing": 0,
        },
        "pairing_rate": Object {
          "incoming": 0,
          "outgoing": 0,
        },
      },
      "id": 1,
      "isExcluded": false,
      "isFlattened": false,
      "link": "http://www.google.de",
      "name": "big leaf",
      "path": "/root/big leaf",
      "type": "File",
    },
  ],
  "descendants": 3,
  "edgeAttributes": Object {
    "average_commits": Object {
      "incoming": 0,
      "outgoing": 0,
    },
    "pairing_rate": Object {
      "incoming": 0,
      "outgoing": 0,
    },
  },
  "id": 0,
  "isExcluded": false,
  "isFlattened": false,
  "name": "root",
  "path": "/root",
  "type": "Folder",
}
`;

exports[`MapTreeViewController onSortingOptionChanged should sort folder structure according to name 1`] = `
Object {
  "attributes": Object {
    "mcc": 111,
    "rloc": 200,
  },
  "children": Array [
    Object {
      "attributes": Object {
        "mcc": 110,
        "rloc": 100,
      },
      "children": Array [
        Object {
          "attributes": Object {
            "functions": 1000,
            "mcc": 10,
            "rloc": 70,
          },
          "edgeAttributes": Object {
            "average_commits": Object {
              "incoming": 0,
              "outgoing": 0,
            },
            "pairing_rate": Object {
              "incoming": 0,
              "outgoing": 0,
            },
          },
          "id": 4,
          "isExcluded": false,
          "isFlattened": false,
          "name": "other small leaf",
          "path": "/root/Parent Leaf/other small leaf",
          "type": "File",
        },
        Object {
          "attributes": Object {
            "functions": 100,
            "mcc": 100,
            "rloc": 30,
          },
          "edgeAttributes": Object {
            "average_commits": Object {
              "incoming": 0,
              "outgoing": 0,
            },
            "pairing_rate": Object {
              "incoming": 0,
              "outgoing": 0,
            },
          },
          "id": 3,
          "isExcluded": false,
          "isFlattened": false,
          "name": "small leaf",
          "path": "/root/Parent Leaf/small leaf",
          "type": "File",
        },
      ],
      "descendants": 2,
      "edgeAttributes": Object {
        "average_commits": Object {
          "incoming": 0,
          "outgoing": 0,
        },
        "pairing_rate": Object {
          "incoming": 0,
          "outgoing": 0,
        },
      },
      "id": 2,
      "isExcluded": false,
      "isFlattened": false,
      "name": "Parent Leaf",
      "path": "/root/Parent Leaf",
      "type": "Folder",
    },
    Object {
      "attributes": Object {
        "functions": 10,
        "mcc": 1,
        "rloc": 100,
      },
      "edgeAttributes": Object {
        "average_commits": Object {
          "incoming": 0,
          "outgoing": 0,
        },
        "pairing_rate": Object {
          "incoming": 0,
          "outgoing": 0,
        },
      },
      "id": 1,
      "isExcluded": false,
      "isFlattened": false,
      "link": "http://www.google.de",
      "name": "big leaf",
      "path": "/root/big leaf",
      "type": "File",
    },
  ],
  "descendants": 3,
  "edgeAttributes": Object {
    "average_commits": Object {
      "incoming": 0,
      "outgoing": 0,
    },
    "pairing_rate": Object {
      "incoming": 0,
      "outgoing": 0,
    },
  },
  "id": 0,
  "isExcluded": false,
  "isFlattened": false,
  "name": "root",
  "path": "/root",
  "type": "Folder",
}
`;

exports[`MapTreeViewController onSortingOptionChanged should sort folder structure according to number of files 1`] = `
Object {
  "attributes": Object {
    "mcc": 111,
    "rloc": 200,
  },
  "children": Array [
    Object {
      "attributes": Object {
        "mcc": 110,
        "rloc": 100,
      },
      "children": Array [
        Object {
          "attributes": Object {
            "functions": 100,
            "mcc": 100,
            "rloc": 30,
          },
          "edgeAttributes": Object {
            "average_commits": Object {
              "incoming": 0,
              "outgoing": 0,
            },
            "pairing_rate": Object {
              "incoming": 0,
              "outgoing": 0,
            },
          },
          "id": 3,
          "isExcluded": false,
          "isFlattened": false,
          "name": "small leaf",
          "path": "/root/Parent Leaf/small leaf",
          "type": "File",
        },
        Object {
          "attributes": Object {
            "functions": 1000,
            "mcc": 10,
            "rloc": 70,
          },
          "edgeAttributes": Object {
            "average_commits": Object {
              "incoming": 0,
              "outgoing": 0,
            },
            "pairing_rate": Object {
              "incoming": 0,
              "outgoing": 0,
            },
          },
          "id": 4,
          "isExcluded": false,
          "isFlattened": false,
          "name": "other small leaf",
          "path": "/root/Parent Leaf/other small leaf",
          "type": "File",
        },
      ],
      "descendants": 2,
      "edgeAttributes": Object {
        "average_commits": Object {
          "incoming": 0,
          "outgoing": 0,
        },
        "pairing_rate": Object {
          "incoming": 0,
          "outgoing": 0,
        },
      },
      "id": 2,
      "isExcluded": false,
      "isFlattened": false,
      "name": "Parent Leaf",
      "path": "/root/Parent Leaf",
      "type": "Folder",
    },
    Object {
      "attributes": Object {
        "functions": 10,
        "mcc": 1,
        "rloc": 100,
      },
      "edgeAttributes": Object {
        "average_commits": Object {
          "incoming": 0,
          "outgoing": 0,
        },
        "pairing_rate": Object {
          "incoming": 0,
          "outgoing": 0,
        },
      },
      "id": 1,
      "isExcluded": false,
      "isFlattened": false,
      "link": "http://www.google.de",
      "name": "big leaf",
      "path": "/root/big leaf",
      "type": "File",
    },
  ],
  "descendants": 3,
  "edgeAttributes": Object {
    "average_commits": Object {
      "incoming": 0,
      "outgoing": 0,
    },
    "pairing_rate": Object {
      "incoming": 0,
      "outgoing": 0,
    },
  },
  "id": 0,
  "isExcluded": false,
  "isFlattened": false,
  "name": "root",
  "path": "/root",
  "type": "Folder",
}
`;

exports[`MapTreeViewController onSortingOrderAscendingChanged should reverse the sorting order 1`] = `
Object {
  "attributes": Object {
    "mcc": 111,
    "rloc": 200,
  },
  "children": Array [
    Object {
      "attributes": Object {
        "mcc": 110,
        "rloc": 100,
      },
      "children": Array [
        Object {
          "attributes": Object {
            "functions": 1000,
            "mcc": 10,
            "rloc": 70,
          },
          "edgeAttributes": Object {
            "average_commits": Object {
              "incoming": 0,
              "outgoing": 0,
            },
            "pairing_rate": Object {
              "incoming": 0,
              "outgoing": 0,
            },
          },
          "id": 4,
          "isExcluded": false,
          "isFlattened": false,
          "name": "other small leaf",
          "path": "/root/Parent Leaf/other small leaf",
          "type": "File",
        },
        Object {
          "attributes": Object {
            "functions": 100,
            "mcc": 100,
            "rloc": 30,
          },
          "edgeAttributes": Object {
            "average_commits": Object {
              "incoming": 0,
              "outgoing": 0,
            },
            "pairing_rate": Object {
              "incoming": 0,
              "outgoing": 0,
            },
          },
          "id": 3,
          "isExcluded": false,
          "isFlattened": false,
          "name": "small leaf",
          "path": "/root/Parent Leaf/small leaf",
          "type": "File",
        },
      ],
      "descendants": 2,
      "edgeAttributes": Object {
        "average_commits": Object {
          "incoming": 0,
          "outgoing": 0,
        },
        "pairing_rate": Object {
          "incoming": 0,
          "outgoing": 0,
        },
      },
      "id": 2,
      "isExcluded": false,
      "isFlattened": false,
      "name": "Parent Leaf",
      "path": "/root/Parent Leaf",
      "type": "Folder",
    },
    Object {
      "attributes": Object {
        "functions": 10,
        "mcc": 1,
        "rloc": 100,
      },
      "edgeAttributes": Object {
        "average_commits": Object {
          "incoming": 0,
          "outgoing": 0,
        },
        "pairing_rate": Object {
          "incoming": 0,
          "outgoing": 0,
        },
      },
      "id": 1,
      "isExcluded": false,
      "isFlattened": false,
      "link": "http://www.google.de",
      "name": "big leaf",
      "path": "/root/big leaf",
      "type": "File",
    },
  ],
  "descendants": 3,
  "edgeAttributes": Object {
    "average_commits": Object {
      "incoming": 0,
      "outgoing": 0,
    },
    "pairing_rate": Object {
      "incoming": 0,
      "outgoing": 0,
    },
  },
  "id": 0,
  "isExcluded": false,
  "isFlattened": false,
  "name": "root",
  "path": "/root",
  "type": "Folder",
}
`;
